worlds/ufo50/games/vainger/rules.py

```diff
@@ -0,0 +1,248 @@
+from typing import TYPE_CHECKING, Dict
+from BaseClasses import Region, CollectionState
+from worlds.generic.Rules import set_rule, CollectionRule
+
+if TYPE_CHECKING:
+    from ... import UFO50World
+
+# adapted from Barbuta, thanks Scipio! <3
+
+# name upgrades for convenience
+heat_mod = "Vainger - Heat Mod"
+multi_mod = "Vainger - Multi Mod"
+pulse_mod = "Vainger - Pulse Mod"
+force_mod = "Vainger - Force Mod"
+
+stabilizer = "Vainger - Stabilizer"
+shield_upgrade = "Vainger - Shield Upgrade"
+keycode_A = "Vainger - Key Code A"
+keycode_B = "Vainger - Key Code B"
+keycode_C = "Vainger - Key Code C"
+keycode_D = "Vainger - Key Code D"
+security_clearance = "Vainger - Progressive Security Clearance"
+
+# can the player do a given hell run?
+# TODO: option for this
+def hell_run(shield_upgrades_required: int, is_vanilla: bool, state: CollectionState, world: UFO50World) -> bool:
+    player = world.player
+    return (state.count(shield_upgrade, player) >= shield_upgrades_required)
```

Comment:
```suggestion
def hell_run(shield_upgrades_required: int, is_vanilla: bool, state: CollectionState, world: "UFO50World") -> bool:
    return state.count(shield_upgrade, world.player) >= shield_upgrades_required
```
UFO50World needs to be in quotes because of weird python typing reasons -- it's the counterpart to that "if TYPE_CHECKING then import UFO50World" thing at the top

`is_vanilla` is unused, though I could see if you were having that there for now until you have options together

return value doesn't need () around it